process/dns_v1.go

```diff
@@ -0,0 +1,336 @@
+package process
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math"
+
+	"github.com/DataDog/mmh3"
+)
+
+// DNS data is encoded as a very basic bucketed hash table.  There are three blocks, or buffers, of data:
+//
+//	The "name" block is all of the unique DNS names.  The length of the name is stored as a varint
+//	followed by the name itself
+//
+//	The "bucket" block contains all of the hash buckets.  The format of each bucket is:
+//		varint for number of entries in bucket
+//		For each entry in the bucket:
+//			varint for length of ip
+//			ip bytes
+//			varint for number of names associated with the ip
+//			Each associated name is encoded as a varint which is the position of the actual name string in the name block
+//
+//	The "position" block is a list of varints, one for each bucket, where each varint is a pointer to the start
+//	of the bucket in the bucket block
+//
+// The overall buffer is encoded as:
+//	1 byte indicating version
+// 	1 byte indicating the number of buckets
```

Comment:
Thoughts on using 2 bytes for this?  341 unique ips would max out the number of keys, and that's not all that many since we're increasing the batch size